tasks/Module 4/task_1.py

```diff
@@ -0,0 +1,27 @@
+"""Mad libs."""
+import os
+import re
+
+TEMPLATE = ['ADJECTIVE', 'NOUN', 'ADVERB', 'VERB']
+
+
+def mad_lib(source_file_path):
+    """Repace placeholders with user inputs."""
+    tmp_file = source_file_path + '.tmp'
+    with open(source_file_path, 'r') as read_file:
+        with open(tmp_file, 'w') as write_file:
+            for line in read_file:
+                pattern = re.compile(f'{"|".join(TEMPLATE)}')
+                matches = pattern.finditer(line)
+                for match in matches:
+                    filler = match.group(0)
+                    replace = input(f'Enter an {filler.lower()}: ')
+                    line = line.replace(filler, replace, 1)
+                print(line)
+                write_file.write(line)
+    os.remove(source_file_path)
+    os.rename(tmp_file, source_file_path)
```

Comment:
I think that it would be more convenient for you if you just create a new text file. It is also in accordance with task description "The results should be printed to the screen and saved to a new text file." ;)

Reply:
done